File: 6. 심화 1/2941.py
string = input()
<<<<<<< HEAD
<<<<<<< HEAD
list_string = []
croatia = ['c=', 'c-', 'dz=', 'd-', 'lj', 'nj', 's=', 'z=']
=======
>>>>>>> 07029c4efeb6747a0fdb1d4d75d8d7f6f9d5c7e1
=======
>>>>>>> 07029c4efeb6747a0fdb1d4d75d8d7f6f9d5c7e1
croatia = ['dz=', 'c=', 'c-', 'd-', 'lj', 'nj', 's=', 'z=']
count = 0

for croa in croatia:
    while croa in string:
        string = string[:string.index(croa)] + ' ' + string[string.index(croa)+len(croa):]
        count += 1
for element in string:
    if (not element == ' ' ) and (element.isalpha()):
        count += 1

<<<<<<< HEAD
<<<<<<< HEAD
list_string = list(string)

while True:
    print(list_string, '출발')
    print(string, '출발')
    for croa in croatia: # 크로아티아에서 i로 하나씩 가져옴
        if croa in string: # 만약 string 안에 i가 있다면 실행
            targets = []
            for target_range in range(len(string)):
                if string[target_range] == croa[0]:
                    targets.append(target_range)
            for target in targets:
                if len(croa) == 2:
                    if string[target+1] == croa[1]:
                        count += 1
                        for del_index in range(target, target+2):
                            list_string[del_index] = ' '
                elif croa == 'dz=':
                    if string[target+1] == croa[1] and string[target+2] == croa[2]:
                        count += 1
                        for del_index in range(target, target+3):
                            list_string[del_index] = ' '
    string = ''
    for i in list_string:
        string += i
    for space in list_string:
        if space != ' ':
if len(string) <= 100:
    for repeat in range(len(string)):
        for croa in croatia:
            if croa in string:
                string = string[:string.index(croa)] + ' ' + string[string.index(croa)+len(croa):]
                count += 1
    for element in string:
        if (not element == ' ') and (element.isalpha()):
            count += 1
    print(count, '결과')
    print(list_string)
    print(string)
    if 



    print(count)
=======
print(count)
>>>>>>> 07029c4efeb6747a0fdb1d4d75d8d7f6f9d5c7e1
=======
print(count)
>>>>>>> 07029c4efeb6747a0fdb1d4d75d8d7f6f9d5c7e1

File: 6. 심화 1/2941.c
#include <stdio.h>
#include <string.h>

int main() {
    char *croatia[8] = {"c=", "c-", "dz=", "d-", "lj", "nj", "s=", "z="};

    char str[101] = {0};
    int count = 0, i;
    scanf("%s", str);

    // for (int i = 0; i < 8; i++) {
    //     for (int j = 0; croatia[i][j] != '\0'; j++) {
    //         printf("%c", croatia[i][j]);
    //     }
    //     printf("\n");
    // }

    for (i = 0; str[i] != '\0'; i++) { // str 글자를 하나씩 가져오기 위함
        // for (int c = i; str[c] != '\0'; c++) {
        //     printf("%c", str[c]);
        // }
        // printf("\n");
        int match = 0;
        for (int j = 0; j < 8; j++) { // croatia 배열에서 문자열 하나씩 가져오기 위함
            if (str[i] == croatia[j][0]) { // 글자하나와 가져온 문자열의 첫글자를 판별해 맞으면 통과
                int result = 1, o;
                for (o  = 1; croatia[j][o] != '\0'; o++) { // 첫글자와 문자열이 맞는지 확인
                    if (str[i+o] != croatia[j][o]) { // 틀리면 다음 문자열로
                        result = 0;
                        break;
                    }
                }
                if (result == 1) { // 만약 맞으면 i는 현재 글자의 인덱스 o는 글자의 수
                    // printf("%d %d %d\n", i, o ,i+o-1);
                    i = i+o-1; // 현재 인덱스랑 글자수를 더해 다음 시작위치 설정
                    count++;
                    match = 1;
                    break;
                }
            }
        }
        if(!match) {count++;}
    }
    
    printf("%d", count);
    return 0;
}